test/utils/test_knowledge_utils.py

```diff
@@ -0,0 +1,357 @@
+#!/usr/bin/env python3
+
+"""
+Test script for world knowledge utilities
+"""
+
+import os
+import numpy as np
+
+from pyrobosim.core import WorldYamlLoader
+from pyrobosim.core import Robot
+from pyrobosim.core import World
+
+# import functions to test
+from pyrobosim.utils.knowledge import (
+    apply_resolution_strategy,
+    query_to_entity,
+    resolve_to_location,
+    resolve_to_object,
+)
+
+from pyrobosim.utils.pose import Pose
+from pyrobosim.utils.general import get_data_folder
+
+
+def load_world():
+    """Load a test world."""
+    world_file = os.path.join(get_data_folder(), "test_world.yaml")
+    return WorldYamlLoader().from_yaml(world_file)
+
+
+def test_apply_resolution_strategy():
```

Comment:
This test is kinda long. Maybe consider splitting the first/random/nearest strategies into separate tests?